From c5bf6c676853a7ea0f67e11fbdb35f53d7fa21b1 Mon Sep 17 00:00:00 2001
From: Terry Zhou <bjzhou@marvell.com>
Date: Tue, 3 Apr 2018 10:05:31 +0800
Subject: [PATCH 1322/1345] staging: mt29f_spinand: disable unsupported
 subpage write function

commit  0e0946416608c12be4a18bbac4c66c893fcfdefb from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

For those drivers without ecc-->hwctl like mt29f, will cause
system crash when calling subpage write which invoke ecc-->hwctl
disable this feature with flag NAND_NO_SUBPAGE_WRITE

Change-Id: I6f4b26eb853d188e7e1ea75ac74a26339c32c789
Signed-off-by: Terry Zhou <bjzhou@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/53206
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Victor Gu <xigu@marvell.com>
Reviewed-by: Nadav Haklai <nadavh@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/staging/mt29f_spinand/mt29f_spinand.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/staging/mt29f_spinand/mt29f_spinand.c b/drivers/staging/mt29f_spinand/mt29f_spinand.c
index e389009..1b79f30 100644
--- a/drivers/staging/mt29f_spinand/mt29f_spinand.c
+++ b/drivers/staging/mt29f_spinand/mt29f_spinand.c
@@ -913,7 +913,7 @@ static int spinand_probe(struct spi_device *spi_nand)
 	chip->read_byte	= spinand_read_byte;
 	chip->cmdfunc	= spinand_cmdfunc;
 	chip->waitfunc	= spinand_wait;
-	chip->options	|= NAND_CACHEPRG;
+	chip->options	|= (NAND_CACHEPRG | NAND_NO_SUBPAGE_WRITE);
 	chip->select_chip = spinand_select_chip;
 
 	mtd = nand_to_mtd(chip);
-- 
1.7.9.5

